test/expect/TestJit.test_alexnet.expect

```diff
@@ -44,22 +44,23 @@ graph(%0 : Double(1, 3, 224, 224)
   %46 : int = prim::Constant[value=9216](), scope: AlexNet
   %47 : int[] = prim::ListConstruct(%45, %46), scope: AlexNet
   %48 : Double(1, 9216) = aten::view(%41, %47), scope: AlexNet
-  %49 : Double(1, 9216) = ^Dropout(0.5, True, False)(%48), scope: AlexNet/Sequential[classifier]/Dropout[0]
-  %50 : Double(9216!, 4096!) = aten::t(%11), scope: AlexNet/Sequential[classifier]/Linear[1]
-  %51 : int = prim::Constant[value=4096](), scope: AlexNet/Sequential[classifier]/Linear[1]
-  %52 : int[] = prim::ListConstruct(%21, %51), scope: AlexNet/Sequential[classifier]/Linear[1]
-  %53 : Double(1, 4096) = aten::expand(%12, %52, %21), scope: AlexNet/Sequential[classifier]/Linear[1]
-  %54 : Double(1, 4096) = aten::addmm(%53, %49, %50, %21, %21), scope: AlexNet/Sequential[classifier]/Linear[1]
-  %55 : Double(1, 4096) = aten::threshold(%54, %23, %23), scope: AlexNet/Sequential[classifier]/ReLU[2]
-  %56 : Double(1, 4096) = ^Dropout(0.5, True, False)(%55), scope: AlexNet/Sequential[classifier]/Dropout[3]
-  %57 : Double(4096!, 4096!) = aten::t(%13), scope: AlexNet/Sequential[classifier]/Linear[4]
-  %58 : Double(1, 4096) = aten::expand(%14, %52, %21), scope: AlexNet/Sequential[classifier]/Linear[4]
-  %59 : Double(1, 4096) = aten::addmm(%58, %56, %57, %21, %21), scope: AlexNet/Sequential[classifier]/Linear[4]
-  %60 : Double(1, 4096) = aten::threshold(%59, %23, %23), scope: AlexNet/Sequential[classifier]/ReLU[5]
-  %61 : Double(4096!, 1000!) = aten::t(%15), scope: AlexNet/Sequential[classifier]/Linear[6]
-  %62 : int = prim::Constant[value=1000](), scope: AlexNet/Sequential[classifier]/Linear[6]
-  %63 : int[] = prim::ListConstruct(%21, %62), scope: AlexNet/Sequential[classifier]/Linear[6]
-  %64 : Double(1, 1000) = aten::expand(%16, %63, %21), scope: AlexNet/Sequential[classifier]/Linear[6]
-  %65 : Double(1, 1000) = aten::addmm(%64, %60, %61, %21, %21), scope: AlexNet/Sequential[classifier]/Linear[6]
-  return (%65);
+  %49 : float = prim::Constant[value=0.5](), scope: AlexNet/Sequential[classifier]/Dropout[0]
+  %50 : Double(1!, 9216) = aten::dropout(%48, %49, %21), scope: AlexNet/Sequential[classifier]/Dropout[0]
```

Comment:
I am confused. I thought jit traces the autograd edges, but why don't you get things like `mul` and `bernoulli_` here? :O

Reply:
Nope, and we don't even want to get those. When we trace autograd, we'll only see the outermost call. In this case `dropout` hijacks the outputs of its internal nodes, and so only its outputs end up being used anywhere. We have a list of ops that skip tracing, because we want to reveal their internals, but this is definitely not the case for dropout.

Reply:
Interesting. Did this logic change recently? I'm asking because when I attempted porting dropout ~1 month ago, it still traced the internal ops (bernoulli outputs as `prim::Constant`, `mul`, etc.). Maybe I remembered wrong?

Reply:
Hmm I think it's been like this for some time now. Can't remember when we added this behavior

Reply:
Okay. Another question then: doesn't this make it harder to optimize across ATen native function boundary?

Reply:
Well it's always easier to lower the high level ops by replacing them with equivalent formulas than to pattern match your way up to higher level abstractions (that's basically the graph isomorphism problem). If we really don't want to trace sth as a high level operator, because there's a potential for optimizations we can add it to the list that skips tracing. I think the current behavior is really what we want.

Reply:
@apaszke Yes of course. Sorry if it sounded that I was questioning the design. I am just wanting to understand it. Thanks for the explanation!